sos/plugins/__init__.py

```diff
@@ -806,13 +808,27 @@ def getmtime(path):
                     self._log_debug("skipping forbidden path '%s'" % _file)
                     continue
                 try:
-                    current_size += os.stat(_file)[stat.ST_SIZE]
+                    filestat = os.stat(_file)
                 except OSError:
                     self._log_info("failed to stat '%s'" % _file)
+                    continue
+                current_size += filestat[stat.ST_SIZE]
+                filetime = datetime.fromtimestamp(filestat[stat.ST_MTIME])
                 if sizelimit and current_size > sizelimit:
                     limit_reached = True
                     break
-                self._add_copy_paths([_file])
+
+                if _file.endswith('.gz') or re.search(r'-\d{8}$', _file):
+                    fileisarchive = True
```

Comment:
A `.gz` isn't the only possible log compression: this should support other formats, or hook into `Policy` to determine the correct value for the system, or a given log file.

Reply:
I've included all the most common compression formats in a compiled regex.